### PR TITLE
add working assemblyscript scaffold

### LINKED ISSUE #26: add hdk-assemblyscript scaffold and hcdev generate option

### PR DESCRIPTION
**Pull request by Connoropolous**
`hcdev g zomes/users assemblyscript`

closes #26 

Based on content from : https://github.com/AssemblyScript/assemblyscript/wiki/Hello-World

### CODE REVIEW

File: src/cli/package.rs [hunk 1]
```diff
@@ -15,6 +15,8 @@ pub const CODE_DIR_NAME: &str = "code";
 
 pub const BUILD_CONFIG_FILE_NAME: &str = ".build";
 
+pub const TSCONFIG_FILE_NAME: &str = "tsconfig.json";
+pub const TYPESCRIPT_FILE_NAME: &str = "index.ts";
 pub const CARGO_FILE_NAME: &str = "Cargo.toml";
```
Comment (sphinxc0re) [inline src/cli/package.rs]: I believe these constants should go into the designated scaffolder

File: src/cli/generate.rs [hunk 2]
```diff
@@ -81,5 +82,13 @@ mod tests {
             .args(&["g", "zomes/bubblechat", "rust"])
             .assert()
             .success();
+
+         Command::main_binary()
+            .unwrap()
+            .current_dir(&tmp.path())
+            .args(&["g", "zomes/zubblebat", "asm"])
```
Comment (sphinxc0re) [inline src/cli/generate.rs]: I'm not sure about the `asm`. I know this is AssemblyScript but `asm` could also mean `asmjs`

**Comment by lucksus** [comment]
I would go with either `as` or `assemblyscript`.

**Comment by Connoropolous** [comment]
will do

**Pull request merged**